### PR TITLE
Should be with options

### PR DESCRIPTION
**Pull request by JakeGinnivan**
Superseeds #322 

@sholland1 and @JoeMighty, thoughts on the changes I have made over the original?

### CODE REVIEW

**Comment by sholland1** [comment]
I was about to make some suggestions...

**Comment by JakeGinnivan** [comment]
I am putting up another xUnit PR based on this, the diff was really big so merged this.

Feel free to make suggestions still and we can refine before release

**Comment by sholland1** [comment]
Ok, cool. I should have some time to work on it tomorrow.
Should I comment on the new PR or this one?

File: src/Shouldly/Internals/Assertions/StringShouldBeAssertion.cs [hunk 1]
```diff
@@ -0,0 +1,55 @@
+﻿using Shouldly.DifferenceHighlighting2;
+using System;
+
+namespace Shouldly.Internals.Assertions
+{
+    internal class StringShouldBeAssertion : IAssertion
```
Comment (JakeGinnivan) [inline src/Shouldly/Internals/Assertions/StringShouldBeAssertion.cs]: Just looking into this a bit more with the xunit changes. Why not use the message generator patterns?

Comment (sholland1) [inline src/Shouldly/Internals/Assertions/StringShouldBeAssertion.cs]: I pretty much explain my thinking here: https://github.com/shouldly/shouldly/issues/302#issuecomment-149386251

**Pull request merged**